geode-core/src/test/java/org/apache/geode/internal/net/NioSslEngineTest.java

```diff
@@ -265,16 +268,22 @@ public void unwrapWithBufferOverflow() throws Exception {
     spyNioSslEngine.engine = testEngine;
 
     testEngine.addReturnResult(
-        new SSLEngineResult(BUFFER_OVERFLOW, NEED_UNWRAP, netBytes.length, netBytes.length),
+        new SSLEngineResult(BUFFER_OVERFLOW, NEED_UNWRAP, 0, 0), // results in 30,000 byte buffer
+        new SSLEngineResult(BUFFER_OVERFLOW, NEED_UNWRAP, 0, 0), // 50,000 bytes
+        new SSLEngineResult(BUFFER_OVERFLOW, NEED_UNWRAP, 0, 0), // 90,000 bytes
         new SSLEngineResult(OK, FINISHED, netBytes.length, netBytes.length));
 
+    int expectedCapacity = 2 * originalPeerAppDataCapacity - originalPeerAppDataPosition;
+    expectedCapacity =
+        2 * (expectedCapacity - originalPeerAppDataPosition) + originalPeerAppDataPosition;
+    expectedCapacity =
+        2 * (expectedCapacity - originalPeerAppDataPosition) + originalPeerAppDataPosition;
     ByteBuffer unwrappedBuffer = spyNioSslEngine.unwrap(wrappedData);
     unwrappedBuffer.flip();
-
-    verify(spyNioSslEngine, times(2)).expandPeerAppData(any(ByteBuffer.class));
-    assertThat(unwrappedBuffer).isEqualTo(ByteBuffer.wrap(netBytes));
+    assertThat(unwrappedBuffer.capacity()).isEqualTo(expectedCapacity);
```

Comment:
i.e. we only check the capacity at the end, but above buffer grows from 30K to 90K bytes...

Reply:
yep